src/librbd/AioImageRequest.cc

```diff
@@ -724,10 +732,140 @@ void AioImageFlush<I>::send_image_cache_request() {
   image_ctx.image_cache->aio_flush(req_comp);
 }
 
+template <typename I>
+void AioImageWriteSame<I>::assemble_extent(const ObjectExtent &object_extent,
+						bufferlist *bl) {
+  uint64_t extent_left = object_extent.length;
```

Comment:
@dillaman here I mean 'extent_left' is how much data this extent can hold and it is directly 'object_extent.length'.
What will 'object_extent.length - object_extent.offset' mean since 'offset' and 'length' is all "in object"? 
Correct me if I was wrong.